From 39efa289a34364e20c250d760c48fd15211f3a08 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Thu, 2 Apr 2015 14:15:34 -0500
Subject: [PATCH] crypto/rand: Disable unsupported readscreen() on Phar Lap ETS

Disable RNG seeding from the screen on Phar Lap ETS (the readscreen()
function). NI's distribution of Phar Lap ETS does not support the
requisite system APIs.

Upstream-Status: Inappropriate [Phar Lap ETS support]

---
 crypto/rand/rand_win.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/rand/rand_win.c b/crypto/rand/rand_win.c
index b4be309..d59cb48 100644
--- a/crypto/rand/rand_win.c
+++ b/crypto/rand/rand_win.c
@@ -678,7 +678,7 @@ static void readtimer(void)
 
 static void readscreen(void)
 {
-# if !defined(OPENSSL_SYS_WINCE) && !defined(OPENSSL_SYS_WIN32_CYGWIN)
+# if !defined(OPENSSL_SYS_WINCE) && !defined(OPENSSL_SYS_WIN32_CYGWIN) && !defined(PHARLAP)
     HDC hScrDC;                 /* screen DC */
     HBITMAP hBitmap;            /* handle for our bitmap */
     BITMAP bm;                  /* bitmap properties */
